article-analyzer/src/main/java/com/example/demo/util/FileHandler.java

```diff
@@ -0,0 +1,47 @@
+package com.example.demo.util;
+
+import com.example.demo.model.Category;
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonParser;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+
+@Component
+@Slf4j
+public class FileHandler {
+
+  public JsonArray getDataFromJsonFile(String filePath) {
+    try (FileReader fileReader = new FileReader(filePath)) {
+      log.info("file read");
+      return JsonParser.parseReader(fileReader).getAsJsonArray();
+    } catch (IOException e) {
+      throw new RuntimeException(e.getMessage());
+    }
+  }
+
+  public void saveDataToFile(Category categories, String categoriesFilePath) {
```

Comment:
I believe this class is intended to be a general-purpose one and has methods that are file agnostic, so it should accept Set of String and file path, but not categories and categoriesFilePath. 
 (what I mean, is the naming of params here should be different)

Also, the method name is too generic, it should be saveSetToJsonFile or something like that

Reply:
updated name of method and names of parameters